spec/reek/cli/command/todo_list_command_spec.rb

```diff
@@ -2,67 +2,95 @@
 
-      it 'shows a proper message' do
-        expected = "\n'.todo.reek' generated! You can now use this as a starting point for your configuration.\n"
-        expect { command.execute }.to output(expected).to_stdout
-      end
+      context 'when does not exist yet' do
+        it 'creates it' do
+          Reek::CLI::Silencer.silently { todo_command.execute }
 
-      it 'returns a success code' do
-        result = Reek::CLI::Silencer.silently { command.execute }
-        expect(result).to eq(Reek::CLI::Status::DEFAULT_SUCCESS_EXIT_CODE)
+          actual_content = File.read(default_configuration_file_name)
+          expect(actual_content).to match(new_configuration_file)
+        end
       end
 
-      it 'writes a todo file with exclusions for each smell' do
-        Reek::CLI::Silencer.silently { command.execute }
-        expected_yaml = {
-          Reek::Configuration::AppConfiguration::DETECTORS_KEY =>
-          { 'UncommunicativeMethodName' => { 'exclude' => ['Smelly#x'] },
-            'UncommunicativeVariableName' => { 'exclude' => ['Smelly#x'] } }
-        }.to_yaml
-        expect(File).to have_received(:write).with(described_class::FILE_NAME, expected_yaml)
-      end
-    end
+      context 'when exists already' do
+        it 'does not update the configuration' do
+          write_existing_configuration default_configuration_file_name
+          command = todo_command configuration: default_configuration
 
-    context 'when no smells re found' do
-      let(:source_file) { CLEAN_FILE }
+          Reek::CLI::Silencer.silently { command.execute }
 
-      it 'shows a proper message' do
-        expected = "\n'.todo.reek' not generated because there were no smells found!\n"
-        expect { command.execute }.to output(expected).to_stdout
+          actual_content = File.read(default_configuration_file_name)
+          expect(actual_content).to match(existing_configuration)
+        end
       end
+    end
 
-      it 'returns a success code' do
-        result = Reek::CLI::Silencer.silently { command.execute }
-        expect(result).to eq Reek::CLI::Status::DEFAULT_SUCCESS_EXIT_CODE
-      end
+    def todo_command(options: Reek::CLI::Options.new([]),
+                     sources: [Pathname.new(SMELLY_FILE.basename.to_s)],
+                     configuration: Reek::Configuration::AppConfiguration.default)
+      described_class.new options: options,
+                          sources: sources,
+                          configuration: configuration
+    end
 
-      it 'does not write a todo file' do
-        Reek::CLI::Silencer.silently { command.execute }
-        expect(File).not_to have_received(:write)
-      end
+    def write_smelly_file
+      File.write(SMELLY_FILE.basename, smelly_file)
+    end
+
+    def write_existing_configuration(file_name)
+      File.write(file_name, existing_configuration)
```

Comment:
I'm not convinced this needs to be a method, since it is only called once. How about inlining it?

Reply:
Ack